### PR TITLE
feature/SOF-7058: python transformation assests

### PR DESCRIPTION
**Pull request by VsevolodX**
The temporary storage for python assets for Materials Designer materials  python transformation capabilities

### CODE REVIEW

File: other/python_transformations/interface_builder.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
+"""Coherent Interface Builder"""
+from __future__ import annotations
+
+# Settings and parameters
```
Comment (timurbazhirov) [inline other/python_transformations/interface_builder.py:4]: Let's put more explanation in a comment

File: other/python_transformations/interface_builder.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+            sub_sl_slab = sub_slab.copy()
+            sub_sl_slab.make_supercell(super_sub_transform)
+            assert_allclose(
+                sub_sl_slab.lattice.matrix[2], sub_slab.lattice.matrix[2], atol=1e-08
+            ), "2D transformation affected C-axis for Film transformation"
+            assert_allclose(
+                sub_sl_slab.lattice.matrix[:2], match.substrate_sl_vectors, atol=1e-08
+            ), "Transformation didn't make proper supercell for substrate"
+
+            # Add extra info
```
Comment (chu-k) [inline other/python_transformations/interface_builder.py:324]: Comment doesn't tell me much. What's in the extra info? Why do we ignore keys that start with '@'?

File: other/python_transformations/interface_builder.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+                "strain": strain,
+                "von_mises_strain": strain.von_mises_strain,
+                "film_sl_vectors": match.film_sl_vectors,
+                "substrate_sl_vectors": match.substrate_sl_vectors,
+                "film_transform": super_film_transform,
+                "substrate_transform": super_sub_transform,
+            }
+
+
+def get_rot_3d_for_2d(film_matrix, sub_matrix) -> np.ndarray:
```
Comment (chu-k) [inline other/python_transformations/interface_builder.py:354]: Where is this used?

File: other/python_transformations/interface_builder.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+    interfaces = list(
+        cib.get_interfaces(
+            terminations[0], gap=INTERFACE_DISTANCE, substrate_thickness=SUBSTRATE_NUMBER_OF_LAYERS, in_layers=True
+        )
+    )
+
+    print("Found {} interfaces".format(len(matches)))
+    print(f"Terminations ({len(terminations)}):", terminations)
+
+    interfaces_list = list(interfaces)
```
[the remark this answers is not in the document]
Comment (VsevolodX) [reply other/python_transformations/interface_builder.py:56]: I believe `interfaces` is an `Iterator`. So it needs to be converted to list

File: other/python_transformations/interface_builder.py [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+from pymatgen.core.surface import SlabGenerator
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator, Sequence
+
+    from pymatgen.core import Structure
+
+
+# Classes and Definitions
+class CoherentInterfaceBuilder:
```
[the remark this answers is not in the document]
Comment (VsevolodX) [reply other/python_transformations/interface_builder.py:126]: The main reason for this class is that I had to Ctrl+C Ctrl+V it from pymatgen to add a little tweak to expose some hidden data. The correct solution is to put this into our Py package, like matera/pymatgen and use here as one-line import.

**Pull request closed without merging**